### ISSUE TITLE
Support setVariables in container component

### ISSUE THREAD

**Issue by raymondsze** [gyzerok/adrenaline#105]
In relay, we are able to use this.props.relay.setVariables to update the variables passed in container component.
Here we have container({ variables, query })(Component)
inside the Component, I can only get the props.mutate and props.isFetching, seems we should have setVariables and variables here which similar to props.relay.variables and props.relay.setVariables.
Are there workaround to solve this issue?

**Comment by gyzerok** [comment]
Right now you can get variables with `this.props.variables`. There is no `this.props.setVariables` and it wouldn't appear in a foreseeable future because I'd like to make decorated components as pure as possible.

In this case you have 2 options:
 * If your component is a presenter component just pass different props to it
 * if your component is a container component then variables that you would like to change are probably part of you route. So to change variable you just need to change route.

**Comment by raymondsze** [comment]
Thanks, here is my approach right now.
I created another wrapper which wrap the adrenaline container as I found that the props passed into container will then pass to its decorated component.
my wrapper has following spec
1. initialVariables
2. prepareVariables
3. query
initialVariables will become the wrapper state so that I could pass setVariables to the adrenaline container to make decorated component to change the wrapper state (i.e variables).
prepareVariables is same as "variables" of adrenaline container, but capture wrapper state instead of props.
query is same as "query" of adrenaline container.
